scripts/typescript/afrl-to-new-api/src/scripts/afrl-csv-to-json.ts

```diff
@@ -0,0 +1,642 @@
+            citation.reference.author = [row.reference];
+
+            // The following does not work, but a better version can work.
+            // I am unsure the reference is always formatted the same way.
+            // citation.reference.author = row.reference.split(',')            // Authors are usually splitted by comas.
+            //                                          .map( a => a.trim() ); // remove empty pre/post spaces
+        }
+
+        // Store in hashmap
+        this.citations.set(row.reference, citation);
+
+        return citation;
+
+    }
+
+    private get_solvent(row: AFRLData): IMaterial {
+
+        const cas = row.solvent_CAS.trim(); // will be used as key in the hashmap this.solvent
+
+        // Try to reuse an existing solvent
+        const existing_solvent = this.solvents.get(cas);
+        if (existing_solvent) {
+            console.log(`-- Found existing solvent: ${existing_solvent.name} (cas: ${cas})`)
+            return existing_solvent
+        }
+
+        // Pull solvent from server
+        // Blocker: the script assumes the solvent already exists on the backend, but we probably don't have it.
+        /*
+        try {
+            solvent = api.get(
+                cript.Material, 
+                {
+                    "identifiers": [
+                        {
+                            "key": "cas", 
+                            "value": cas
+                        }
+                    ], 
+                    "project": cript_project.uid
+                },
+                max_level=0
+            )
+            print(f"ROW {index + 2} -- Found existing solvent: {solvent.name}")
+            solvents[cas] = solvent
+            return solvent
+        except (cript.exceptions.APIGetError):
+            return None
+        */
+
+        // Temporary solution: we create a new Solvent...        
+        const solvent: IMaterial = {
+            node: ['Material'],
+            name: row.solvent,
+            cas,
+            model_version: MODEL_VERSION,
+        } as IMaterial;
+        this.record_error(`Search material from "cas" is not implemented, creating a local solvent for ${JSON.stringify(solvent)}`)
+
+        // Store in hashmap
+        this.solvents.set(cas, solvent);
+
+        return solvent;
+    }
+
+    private get_polymer(row: AFRLData, citation: ICitation[] = []): IMaterial {
+
+        const polymer_id = row.polymer_id;
+        const name = row.polymer;
+        const unique_name = `${name}_${polymer_id}`;
+        const cas = row.polymer_CAS;
+        const bigsmiles = this.smiles_to_BigSMILES(row.polymer_SMILES)
+        const mw_w = row.polymer_Mw;
+        const mw_d = row.polymer_PDI;
+
+        // Try to get the existing
+        if (citation) {
+            // Note: previous implementation was using both mw_w, mw_d, and name as hash, but using unique_name seems more appropriate.
+            const existing_polymer = this.polymers.get(unique_name);
```

Comment:
I am unsure of that now. Should I use the `cas` as keys in `this;polymer` map ?
I case a change is required here, update line 281 accordingly.

Reply:
From what I remember `CAS` was renamed to `chemical_id` for our platform

Reply:
No, there is cas. But we cannot search by cas.

Reply:
it should have been replaced with `chemical_id` because `CAS` is unavailable to our platform

Reply:
You are correct. The vocabulary do not mention `cas` but do mention `chemical_id`. I'll still use `cas` in the code because the source data uses this concept.
Unfortunately I cannot store the `cas` as a `chemical_id` on a material with the current schema. Such a field does not exists. Since `cas` is present on the schema I fallback on this field temporarly:

![image](https://user-images.githubusercontent.com/111717478/230660195-fb616664-467d-45e7-9d31-56ff70e4638a.png)